document-portal/document-portal-fuse.c

```diff
@@ -3155,28 +3283,33 @@ xdp_fuse_init (GError **error)
       return FALSE;
     }
 
-  main_ch = fuse_mount (path, &args);
-  if (main_ch == NULL)
+  if (fuse_parse_cmdline (&args, &opts) != 0)
     {
       fuse_opt_free_args (&args);
-      g_set_error (error, XDG_DESKTOP_PORTAL_ERROR, XDG_DESKTOP_PORTAL_ERROR_FAILED,
-                   "Can't mount fuse fs on %s: %s", path, g_strerror (errno));
+      g_set_error (error, XDG_DESKTOP_PORTAL_ERROR,
+                   XDG_DESKTOP_PORTAL_ERROR_INVALID_ARGUMENT,
+                   "Impossible to parse command line");
       return FALSE;
     }
 
-  session = fuse_lowlevel_new (&args, &xdp_fuse_oper,
-                               sizeof (xdp_fuse_oper), NULL);
-  if (session == NULL)
+  thread_data.args = &args;
+  thread_data.loop_config.clone_fd = opts.clone_fd;
+  thread_data.loop_config.max_idle_threads = opts.max_idle_threads;
+
+  fuse_thread = g_thread_new ("fuse mainloop", xdp_fuse_thread, &thread_data);
+
+  while (g_atomic_pointer_get (&session) == NULL &&
+         g_atomic_pointer_get (&thread_data.error) == NULL)
+    g_main_context_iteration (NULL, FALSE);
```

Comment:
Is it actually desirable to process the event loop here? I'd have thought we'd want to block until the thread initialises. If we progress the event loop, we might receive more D-Bus method calls that expect the fuse thread to be ready.

Reply:
Mh that's true, IIRC such calls should not be replied anyways but I agree would be better that.